### PR TITLE
Actions for getting uniswap v3 pool contract address, liquidity, slot0 data, and observation data

### PR DESCRIPTION
**Pull request by rohan-agarwal-coinbase**
4 new actions for reading data for deployed uniswap liquidity pools (pool contract address, liquidity, slot0, observation data based on seconds ago)

### Testing
- Unit tests added
- Tested E2E to see that we're correctly fetching the pool details for the 4 new read actions

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: cdp-agentkit-core/cdp_agentkit_core/actions/uniswap_v3/get_pool.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+from pydantic import BaseModel, Field
+
+from cdp_agentkit_core.actions.cdp_action import CdpAction
+from cdp_agentkit_core.actions.uniswap_v3.constants import (
+    UNISWAP_V3_FACTORY_ABI,
+    get_contract_address,
+)
+
+UNISWAP_V3_GET_POOL_PROMPT = """
+This tool will get the pool contract address for Uniswap V3 pools that have been previously created. It takes in the networkId, the two token addresses, with the value 0x4200000000000000000000000000000000000006 for native gas token and a value like 0x1234567890123456789012345678901234567890 for ERC20 token. It also takes in the pool fee, which is denominated in hundredths of a bip (i.e. 1e-6). Acceptable fee values are 100, 500, 3000, and 10000. Supported networks are Base Sepolia, Base Mainnet, Ethereum Mainnet, Polygon Mainnet, and Arbitrum Mainnet.
```
Comment (John-peterson-coinbase) [inline cdp-agentkit-core/cdp_agentkit_core/actions/uniswap_v3/get_pool.py:13]:
```suggestion
This tool will get the pool contract address for Uniswap V3 pools that have been previously created. It takes in the network ID, the two token addresses, with the value 0x4200000000000000000000000000000000000006 for native gas token and a value like 0x1234567890123456789012345678901234567890 for ERC20 token. It also takes in the pool fee, which is denominated in hundredths of a bip (i.e. 1e-6). Acceptable fee values are 100, 500, 3000, and 10000. Supported networks are Base Sepolia, Base Mainnet, Ethereum Mainnet, Polygon Mainnet, and Arbitrum Mainnet.
```

File: cdp-agentkit-core/cdp_agentkit_core/actions/uniswap_v3/get_pool_liquidity.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+from cdp import SmartContract
+from pydantic import BaseModel, Field
+
+from cdp_agentkit_core.actions.cdp_action import CdpAction
+from cdp_agentkit_core.actions.uniswap_v3.constants import (
+    UNISWAP_V3_POOL_ABI,
+)
+
+UNISWAP_V3_GET_POOL_LIQUIDITY_PROMPT = """
+This tool will get the liquidity for Uniswap V3 pools that have been previously created. It takes in the networkId and the pool contract address. Supported networks are Base Sepolia, Base Mainnet, Ethereum Mainnet, Polygon Mainnet, and Arbitrum Mainnet.
```
Comment (John-peterson-coinbase) [inline cdp-agentkit-core/cdp_agentkit_core/actions/uniswap_v3/get_pool_liquidity.py:12]:
```suggestion
This tool will get the liquidity for Uniswap V3 pools that have been previously created. It takes in the network ID and the pool contract address. Supported networks are Base Sepolia, Base Mainnet, Ethereum Mainnet, Polygon Mainnet, and Arbitrum Mainnet.
```

File: cdp-agentkit-core/cdp_agentkit_core/actions/uniswap_v3/get_pool_observe.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+from cdp import SmartContract
+from pydantic import BaseModel, Field
+
+from cdp_agentkit_core.actions.cdp_action import CdpAction
+from cdp_agentkit_core.actions.uniswap_v3.constants import (
+    UNISWAP_V3_POOL_ABI,
+)
+
+UNISWAP_V3_GET_POOL_OBSERVE_PROMPT = """
+This tool will get the observation details for Uniswap V3 pools that have been previously created. It takes in a networkId, pool contract address, and list of secondsAgo values and returns the tickCumulative and secondsPerLiquidityCumulativeX128 for each secondsAgo value.
```
Comment (John-peterson-coinbase) [inline cdp-agentkit-core/cdp_agentkit_core/actions/uniswap_v3/get_pool_observe.py:12]:
```suggestion
This tool will get the observation details for Uniswap V3 pools that have been previously created. It takes in a network ID, pool contract address, and list of secondsAgo values and returns the tickCumulative and secondsPerLiquidityCumulativeX128 for each secondsAgo value.
```

File: cdp-agentkit-core/cdp_agentkit_core/actions/uniswap_v3/get_pool_slot0.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+from cdp import SmartContract
+from pydantic import BaseModel, Field
+
+from cdp_agentkit_core.actions.cdp_action import CdpAction
+from cdp_agentkit_core.actions.uniswap_v3.constants import (
+    UNISWAP_V3_POOL_ABI,
+)
+
+UNISWAP_V3_GET_POOL_SLOT0_PROMPT = """
```
Comment (John-peterson-coinbase) [inline cdp-agentkit-core/cdp_agentkit_core/actions/uniswap_v3/get_pool_slot0.py:11]: Please specify it takes network ID and pool contract address as input

**Review by John-peterson-coinbase — APPROVED** [review]

**Pull request merged**